transaction-history-manager/src/main/java/com/qpay/transactionhistorymanager/service/impl/ReportGeneratorIText.java

```diff
@@ -0,0 +1,203 @@
+        final List<TransactionEntity> transactionEntitiesInPeriod = getTransactionEntitiesInPeriod(reportInfo);
+        for (int number = 0; number < transactionEntitiesInPeriod.size(); number++) {
+            final var transactionEntity = transactionEntitiesInPeriod.get(number);
+            final BigDecimal amount = transactionEntity.getAmount();
+            final var transactionType = transactionEntity.getTransactionType();
+            transactionsTable.addCell(String.valueOf(number + 1));
+            transactionsTable.addCell(transactionEntity.getCreatedAt().format(DATE_PATTERN_dd_MM_yyyy_HH_mm));
+            transactionsTable.addCell(transactionType.toString());
+            transactionsTable.addCell(amount.toString());
+            transactionsTable.addCell(transactionEntity.getNameTo());
+            if (reportInfo.userType() == UserType.CUSTOMER) {
+                if (transactionType == TransactionType.PAYMENT) {
+                    totalSent = totalSent.add(amount);
+                } else if (transactionType == TransactionType.TOP_UP) {
+                    totalReceived = totalReceived.add(amount);
+                }
+            } else {
+                totalReceived = totalReceived.add(amount);
+            }
+        }
+        final List<Paragraph> paragraphList = List.of(
+                createLogoParagraph(),
+                createTitleParagraph(),
+                createSubTitleParagraph(reportInfo),
+                createClientNameParagraph(customerReportInfo.name()),
+                createTotalAmountParagraph("income", totalReceived),
+                createTotalAmountParagraph("expenses", totalSent),
+                new Paragraph().add(transactionsTable)
+        );
+        savePdf(path, paragraphList);
+    }
+
+    private List<TransactionEntity> getTransactionEntitiesInPeriod(final ReportInfo reportInfo) {
+        if (reportInfo.userType() == UserType.CUSTOMER) {
+            return transactionRepository.findAllByIdFromAndCreatedAtBetween(
+                            reportInfo.userId(),
+                            reportInfo.periodStart(),
+                            reportInfo.periodEnd()
+                    );
+        } else {
+            return transactionRepository.findAllByIdToAndCreatedAtBetween(
+                            reportInfo.userId(),
+                            reportInfo.periodStart(),
+                            reportInfo.periodEnd()
+                    );
+        }
+    }
+
+    void savePdf(final String path, final List<Paragraph> paragraphList) {
+        try {
+            FileManager.savePdf(path, paragraphList);
+        } catch (final IOException exception) {
+            throw new SavePdfException(exception.getMessage());
+        }
+    }
+
+    private String generatePath(final long userId) {
+        final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd:hh:mm:ss");
+        final String currentDateTime = dateFormat.format(new Date());
```

Comment:
For each PR I write everywhere to use `var`. Do it yourself, I'm tired of writing it for every single case